dhall/src/Dhall/Import.hs

```diff
@@ -744,8 +748,55 @@ toHeader _ = do
     empty
 
 getCacheFile
-    :: (Alternative m, MonadIO m) => FilePath -> Dhall.Crypto.SHA256Digest -> m FilePath
+    :: (MonadCatch m, Alternative m, MonadIO m) => FilePath -> Dhall.Crypto.SHA256Digest -> m FilePath
 getCacheFile cacheName hash = do
+    cacheDirectory <- getOrCreateCacheDirectory cacheName
+
+    let cacheFile = cacheDirectory </> ("1220" <> show hash)
+
+    return cacheFile
+
+warnAboutMissingCaches  :: (MonadCatch m, Alternative m, MonadIO m) => m ()
+warnAboutMissingCaches = do
+    warnings <- traverse warnAboutMissingCache ["dhall", "dhall-haskell"]
+
+    case catMaybes warnings of
+        warn:_ -> liftIO (System.IO.hPutStrLn System.IO.stderr warn)
+        []     -> return () 
+      
+
+warnAboutMissingCache :: (MonadCatch m, Alternative m, MonadIO m) => FilePath -> m (Maybe String)
+warnAboutMissingCache cacheName = do
+    mbCacheBaseDir <- Maybe.runMaybeT getCacheBaseDirectory
+
+    case mbCacheBaseDir of
+        Just cacheBaseDir -> do
+            let expectedCacheDir = cacheBaseDir </> cacheName
+
+            mbCacheDir <- 
+                Maybe.runMaybeT (getOrCreateCacheDirectory cacheName)
+
+            case mbCacheDir of
+                Just _ -> return Nothing
+                
+                Nothing -> return (Just (warningCacheDir expectedCacheDir))
+
+        Nothing -> return (Just warningEnvVar)
+ 
+    where warningEnvVar = 
```

Comment:
You can simplify this by pushing these warnings further down into the `get*` functions (i.e. push `warningEnvVar` into the `getCacheBaseDirectory` and pushing `warningCacheDir` into the `getOrCreateCacheDirectory` function).  Then you don't need separate top-level functions for warnings.

Reply:
Mmm, my idea was not touch the `get*` function cause it is called by `getCacheFile` and the message would be shown several times (one for each hashed import, right?)

So those functions [have to change](https://github.com/dhall-lang/dhall-haskell/pull/1320#issuecomment-532600886), adding a parameter to activate warnings or returning an `Either Warning Path`. Not sure if it worths it, but what would be better? (or maybe i miss some another option)

Reply:
I think you could still create a `getOrCreateOrWarnCacheDirectory` utility which wraps `getOrCreateCacheDirectory`

Reply:
Thanks for the suggestion!, this way i can leverage the `Alternative` class